DataFormats/Headers/include/Headers/DataHeader.h

```diff
@@ -215,33 +201,41 @@ constexpr T String2(const char (&str)[N])
 /// have to be studied.
 template <int N, typename PrinterPolicy = Internal::defaultPrinter>
 struct Descriptor {
-  static_assert(Internal::NumberOfActiveBits<N>::value == 1 && N <= 8,
+  static_assert(Internal::NumberOfActiveBits<N>::value == 1 && !(N%2),
 		"size is required to be 1 or a multiple of 2");
   static int const size = N;
   static int const bitcount = size*8;
-  static int const arraySize = 1; //Internal::ArraySize<size, uint64_t>::value;
+  static constexpr int arraySize = Internal::ArraySize<uint64_t, size>();
   using ItgType = typename Internal::TraitsIntType<N>::Type;
 
   union {
     char     str[N];
-    ItgType  itg; // for extension > 64 bit: [arraySize];
+    ItgType  itg[arraySize];
   };
-  constexpr Descriptor() : itg(0) {};
-  constexpr Descriptor(ItgType initializer) : itg(initializer) {};
-  constexpr Descriptor(const Descriptor& other) : itg(other.itg) {}
-  Descriptor& operator=(const Descriptor& other) {
-    if (&other != this) itg = other.itg;
-    return *this;
-  }
+  constexpr Descriptor() : itg{0} {};
+  constexpr Descriptor(ItgType initializer) : itg{initializer} {};
+  constexpr Descriptor(const Descriptor& other) = default;
+  Descriptor& operator=(const Descriptor& other) = default;
+
   // Note: don't need to define operator=(ItgType v) because the compiler
   // can use Descriptor(ItgType initializer) for conversion
 
   // type cast operator for simplified usage of the descriptor's integer member
-  operator ItgType() const {return itg;}
+  // TODO: this is sort of a hack, takes the first element.
+  //       we should rethink these implicit conversions
+  operator ItgType() const {return itg[0];}
 
   /// constructor from a compile-time string
-  template<std::size_t NN>
-  constexpr Descriptor(const char (&origin)[NN]) : itg(String2<ItgType>(origin)) {};
+  constexpr Descriptor(const char* in) : str{0}
+  {
+    int i = 0;
+    for (; in[i] && i < N ; ++i) {
+      str[i] = in[i];
+    }
+  }
+
+  template <int M>
+  friend constexpr Descriptor<M> DescriptorInit(const char* in);
```

Comment:
yes, that is part of the refactoring, chunk included by mistake. Will move to next PR.
Idea is (obviously) to make Descriptor(const char*) private and make init awkward (and possible only from special data definition headers)